mlir/lib/Dialect/Rock/Generator/ConvGenerator.cpp

```diff
@@ -654,30 +654,38 @@ ConvGenerator::parseConvDims(int64_t batchSize, int64_t groupSize,
                              int64_t filterHeight, int64_t filterWidth) {
   config.filterDims[DIM::HEIGHT] = filterHeight;
   config.filterDims[DIM::WIDTH] = filterWidth;
-  static const std::string filterKeys = "kgcyx";
-  int64_t filterVals[] = {outputChannel / groupSize, groupSize,
-                          inputChannel / groupSize, filterHeight, filterWidth};
-
-  static const std::string inputKeys = "ngchw";
-  int64_t inputVals[] = {batchSize, groupSize, inputChannel / groupSize,
-                         inputHeight, inputWidth};
-
-  static const std::string outputKeys = "ngkhw";
-  int64_t outputVals[] = {batchSize, groupSize, outputChannel / groupSize,
-                          outputHeight, outputWidth};
-
-  auto convertLayout = [](char &key, const std::string &kmap, int64_t vals[],
-                          auto &dims) {
-    auto keyl = std::tolower(key);
-    auto ii = kmap.find(keyl);
-    if (ii == std::string::npos) {
-      static std::string nchw = "ngchw";
-      ii = nchw.find(keyl);
-      if (ii == std::string::npos)
+
+  llvm::StringMap<int64_t> filterMap = {{"k", outputChannel / groupSize},
+                                        {"g", groupSize},
+                                        {"c", inputChannel / groupSize},
+                                        {"y", filterHeight},
+                                        {"x", filterWidth},
+                                        {"0", filterHeight},
```

Comment:
Both old and new keys are present in the map, for now.

Reply:
This is a nice cleanup